File: rag_ui/frontend/src/types/topics.ts
export interface UserTopic {
  name: string;
  wiki_title: string;
  added_at: string;
  chunks_count: number;
}

export interface TopicsAvailableResponse {
  default_topics: string[];
  user_topics: UserTopic[];
  all_topics: string[];
}

export interface AddTopicResponse {
  message: string;
  chunks_added: number;
  wiki_title: string;
}

File: backend/models/topics.py
from pydantic import BaseModel
from typing import List


class AddTopicRequest(BaseModel):
    topic: str


class TopicEntry(BaseModel):
    name: str
    wiki_title: str
    added_at: str
    chunks_count: int


class AvailableTopicsResponse(BaseModel):
    default_topics: List[str]
    user_topics: List[TopicEntry]
    all_topics: List[str]
